securedrop/utils.py

```diff
@@ -0,0 +1,21 @@
+from itertools import count
+from logging import Logger
+from typing import Callable
+
+from jinja2.runtime import Context
+from jinja2.utils import pass_context
+
+
+def create_get_descriptive_title_function(logger: Logger) -> Callable[[], str]:
+    @pass_context
+    def get_descriptive_title(context: Context) -> str:
```

Comment:
The [latest `mypy` failure](https://app.circleci.com/pipelines/github/freedomofpress/securedrop/5355/workflows/0c5e0b21-a8d6-41c5-a2a3-f5096745ce4b/jobs/66870?invite=true#step-110-451) is legit. The type hint for the return value does indeed not match the signature of the returned function.

```suggestion
def create_get_descriptive_title_function(logger: Logger) -> Callable[[Context], str]:
    @pass_context
    def get_descriptive_title(context: Context) -> str:
```

Reply:
As you noted below, this actually gets passed in automatically by jinja, so the type hint is correct. I just uploaded a new commit with a comment telling mypy to ignore this error and explaining the rationale.